Sources/Brave/Frontend/Browser/BrowserViewController.swift

```diff
@@ -1820,7 +1822,7 @@ public class BrowserViewController: UIViewController {
       
       guard let scheme = tab.webView?.url?.scheme,
             let host = tab.webView?.url?.host else {
-        tab.secureContentState = .insecure
+        tab.secureContentState = .unsupportedProtocol
```

Comment:
can you explain the logic here? what's an example of a URL that would trigger unsupported protocol? in the spec the example was https://dh480.badssl.com/

Reply:
I haven't been able to find a site that will trigger this, the example of dh480 actually results in a cert error in our path because WebKit reports it as not having a server trust and the error becomes `NSURLErrorSecureConnectionFailed`, so we don't have the same pathway as desktop for this case.

Reply:
in that case maybe we should just combine this with one of the other error states? if there's no way for QA to test this is working i'm not sure we should even have it.

Reply:
Even prior to this I dont think there was really any way of testing this since it relies on the scheme or host of a URL being empty, which wouldn't result in a valid url to load anyways so I can remove this case entirely if we'd like yeah

Reply:
I believe "internal" URL will trigger this as it doesn't have a cert but it's our internal pages

Reply:
we probably don't want the internal pages to show up with any icon? they are essentially secure since they're usually just loading local content.

not sure what to show for reader mode on an HTTP page though

Reply:
Internal pages wont trigger this because they're handled separately just above this in the server trust guard failure case. Error pages get the insecure state based on if its an cert error page or not, reader mode gets `localhost`  which right now shows no icons (technically reader mode is an `internal://` URL)

Reply:
I think after https://github.com/brave/brave-ios/commit/c07cc052fff2beb63b7e1d692deabaf233fbf46f reader mode now gets a warning icon, there is also no button to exit reader mode:![image](https://github.com/brave/brave-ios/assets/10810135/d5218e2f-dc06-4d6e-8b24-0e9b683f5652)

Reply:
@stoletheminerals Good catch, not sure what went wrong there, will fix!

Reply:
Fixed in https://github.com/brave/brave-ios/pull/8417/commits/ee6710fa7f8d5c7262e6337a7635dc1d6be39c59